google/cloud/bigtable/data/execute_query/_async/execute_query_iterator.py

```diff
@@ -0,0 +1,169 @@
+        self._result_generator = self._next_impl()
+        self._register_instance_task = None
+        self._is_closed = False
+        self._request_body = request_body
+        self._attempt_timeout_gen = _attempt_timeout_generator(
+            attempt_timeout, operation_timeout
+        )
+        self._async_stream = retries.retry_target_stream_async(
+            self._make_request_with_resume_token,
+            retries.if_exception_type(*retryable_excs),
+            retries.exponential_sleep_generator(0.01, 60, multiplier=2),
+            operation_timeout,
+            exception_factory=_retry_exception_factory,
+        )
+        self._req_metadata = req_metadata
+
+        try:
+            self._register_instance_task = asyncio.create_task(
+                self._client._register_instance(instance_id, self)
+            )
+        except RuntimeError as e:
+            raise RuntimeError(
+                f"{self.__class__.__name__} must be created within an async event loop context."
+            ) from e
+
+    @property
+    def is_closed(self):
+        return self._is_closed
+
+    @property
+    def app_profile_id(self):
+        return self._app_profile_id
+
+    @property
+    def table_name(self):
+        return self._table_name
+
+    async def _make_request_with_resume_token(self):
+        resume_token = self._byte_cursor.prepare_for_new_request()
+        request = ExecuteQueryRequestPB(
+            {
+                **self._request_body,
+                "resume_token": resume_token,
+            }
+        )
+        return await self._client._gapic_client.execute_query(
+            request,
+            timeout=next(self._attempt_timeout_gen),
+            metadata=self._req_metadata,
+            retry=None,
+        )
+
+    async def _await_metadata(self) -> None:
+        if self._byte_cursor.metadata is None:
+            metadata_msg = await self._async_stream.__anext__()
+            self._byte_cursor.consume_metadata(metadata_msg)
+
+    async def _next_impl(self) -> AsyncIterator[QueryResultRow]:
+        await self._await_metadata()
+
+        async for response in self._async_stream:
+            try:
+                bytes_to_parse = self._byte_cursor.consume(response)
+                if bytes_to_parse is None:
+                    continue
+
+                results = self._reader.consume(bytes_to_parse)
+                if results is None:
+                    continue
+
+            except ValueError as e:
+                raise InvalidExecuteQueryResponse(
+                    "Invalid ExecuteQuery response received"
+                ) from e
+
+            for result in results:
+                yield result
+        await self.close()
+
+    async def __anext__(self):
+        if self._is_closed:
+            raise StopAsyncIteration
+        return await self._result_generator.__anext__()
+
+    def __aiter__(self):
+        return self
+
+    async def metadata(self) -> Optional[Metadata]:
```

Comment:
please add docstrings on all of these methods

Reply:
done